### PR TITLE
Clean out non-gfs top level variables

### LINKED ISSUE #2332: Clean out non-gfs top level variables
### What new functionality do you need?

Removal of duplicate variables that define the top level folder locations (e.g. `PARMwave` is the same as `PARMgfs`).

This is related to cleanup that happened in issue #2184 for the `FIX*` variables.

Remove the non-gfs versions of `PARM*`, `USH*`, `UTIL*`, `EXEC*`, `SCR*`. Also remove `*gfs` variables outside of configs including `HOME*` variables and `EXPDIR`s that aren't needed.

### What are the requirements for the new functionality?

Functionality unimpacted.

### Acceptance Criteria

The only top level path variables are now `*gfs` versions.

### Suggest a solution (optional)

_No response_

### PR DESCRIPTION
**Pull request by KateFriedman-NOAA**
# Description

This PR cleans out non-gfs top level variables that are duplicates or no longer needed. This also standardizes how we set these variables in scripts. Changes touch much of the system but do not change functionality. Changes include:

- Remove/retire variables such as `PARMwmo`, `EXECwave`, `USHgempak`, `HOMEbufrsnd` and numerous additional versions of those
- Remove top-level variable settings/exports from scripts (should only be set in configs now, mainly `config.base`)
- Replace instances of `HOMEgfs/parm` with `PARMgfs` and other similar variables
- Retire `UTILgfs`; no more `/util` folder at top level
- Change `USHgempak` to `HOMEgfs/gempak/ush`
- Replace `IODAPARM` with `PARMgfs/gdas/bufr2ioda` (discussed this and related changes with @RussTreadon-NOAA )
  - Related to this, add `bufr2ioda` parm file linking to link script
- Remove unnecessary script variables and replace with actual path and name of script (e.g. `AWIPS20KM1P0DEGSH` and `AWIPSG2SH`)
- Remove unnecessary exec variable with top level variable (`FCSTEXECDIR` -> `EXECgfs`)
- Correct bug found with incorrectly named variable for `/scripts` folder (`SRCgfs` -> `SCRgfs`)
- Some cleanup: Remove `fix/gldas` and `parm/gldas` from `.gitignore` and add `parm/gdas/bufr2ioda`
- Remove an unused function to clean the GSI monitor outputs for space management. Discussed with @EdwardSafford-NOAA before removing.

Had also intended to clean out some unnecessary checks for variables being set but will do that in a separate issue.

Resolves #2332

# Type of change

- Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?

Low-res atmos-only cycled test on WCOSS2 with all `DO_*` variables turned on. Some jobs failed (expected) but I checked the logs to make sure my changes weren't the cause.

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Review by RussTreadon-NOAA — APPROVED** [review]
Changes related to `prepatmiodaobs` look correct.  Approve for `prepatmiodaobs` changes. 

Conditional approval pending (a) approval from other subject matter experts and (b) successful CI tests.

**Review by EdwardSafford-NOAA — APPROVED** [review]
Changes to DA monitor scripts and config files look good.  Approved, pending approval of other subject matter experts.

**Review by HenryRWinterbottom — APPROVED** [review]
LGTM. Conditionally approving pending other reviewer comments/suggestions and successful CI tests.

**Review by JessicaMeixner-NOAA — APPROVED** [review]
Wave parts look okay to me

File: scripts/exgdas_atmos_verfrad.sh
Comment (WalterKolczynski-NOAA) [inline scripts/exgdas_atmos_verfrad.sh:142]: I didn't see this change mentioned in the PR.

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]
None of the preamble lines got updated to use `${USHgfs}`

File: scripts/exgdas_atmos_verfrad.sh
Comment (KateFriedman-NOAA) [reply scripts/exgdas_atmos_verfrad.sh:142]: Apologies, good catch. I have updated the PR description to include this change. Thanks!

**Comment by KateFriedman-NOAA** [comment]
> None of the preamble lines got updated to use `${USHgfs}`

My bad, those should all get updated too! I will do that and run a quick test on WCOSS2 to make sure all is well. Will ping you for a rereview when done. Thanks @WalterKolczynski-NOAA !

**Comment by KateFriedman-NOAA** [comment]
> > None of the preamble lines got updated to use `${USHgfs}`
> 
> My bad, those should all get updated too! I will do that and run a quick test on WCOSS2 to make sure all is well. Will ping you for a rereview when done. Thanks @WalterKolczynski-NOAA !

@WalterKolczynski-NOAA So I can't change the preamble or jjob_header lines to use `USHgfs` because `USHgfs` isn't know at that point yet. It isn't exported until `config.base` is sourced. `HOMEgfs` is known via being exported from the rocoto xml or ecflow. I do see some other places where `HOMEgfs/something` can be replaced with one of the other top level variables though. I will scan through again and change those but leave the preamble and jjob_header sourcing as is.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

File: jobs/JGFS_ATMOS_GEMPAK_PGRB2_SPEC
Comment (aerorahul) [inline jobs/JGFS_ATMOS_GEMPAK_PGRB2_SPEC:19]: ??
Should this be in a `config.gempak_spec` file?

File: ush/fv3gfs_remap.sh
Comment (aerorahul) [inline ush/fv3gfs_remap.sh]: I don't think we use this script anymore.

**Review by aerorahul — APPROVED** [review]
looks good.
just spotted a few curious items.

File: jobs/JGFS_ATMOS_GEMPAK_PGRB2_SPEC
Comment (KateFriedman-NOAA) [reply jobs/JGFS_ATMOS_GEMPAK_PGRB2_SPEC:19]: Probably. @WalterKolczynski-NOAA is this something you could pull out of this script and into a config within your gempak work?

File: ush/fv3gfs_remap.sh
Comment (KateFriedman-NOAA) [reply ush/fv3gfs_remap.sh]: You're right, we don't. I just searched for it within the system and did not find any. I will remove this script.

Comment (KateFriedman-NOAA) [reply ush/fv3gfs_remap.sh]: Script has been sent to the farm @ e8136b60

**Pull request merged**
